core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -433,7 +434,7 @@ public void run() {
               observer.onHeaders(headers);
             } catch (Throwable t) {
               cancel();
-              throw Throwables.propagate(t);
+              Throwables.propagate(t);
```

Comment:
When writing the code, I'd rather not defeat Java's dead code and return detection. I think this is more a code style thing that should be left as-is unless there is a need to change it. So in new code you can do it either way, but let's not change existing code unless need arises.